internal/scm/repository.go

```diff
@@ -9,3 +9,18 @@ type Repository interface {
 	// FullName returns the full id of the repository, usually ownerName/repoName
 	FullName() string
 }
+
+func RepoContainsTopic(slice1 []string, slice2 []string) bool {
+	slice1Map := map[string]struct{}{}
+	for _, v := range slice1 {
+		slice1Map[v] = struct{}{}
+	}
+
+	for _, v := range slice2 {
+		if _, ok := slice1Map[v]; ok {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
We can now add more precise names to the arguments.
```suggestion
func RepoContainsTopic(repoTopics []string, filterTopics []string) bool {
	repoTopicsMap := map[string]struct{}{}
	for _, v := range repoTopics {
		repoTopicsMap[v] = struct{}{}
	}

	for _, v := range filterTopics {
		if _, ok := repoTopicsMap[v]; ok {
			return true
		}
	}

	return false
}
```